pkg/aws/core.go

```diff
@@ -171,6 +178,44 @@ func (d *Driver) CreateMachine(ctx context.Context, req *driver.CreateMachineReq
 	return response, nil
 }
 
+//returns Placement Object required in ec2.RunInstancesInput
+func getPlacementObj(req *driver.CreateMachineRequest) *ec2.Placement {
+
+	var tenancy string = ""
+	var affinity string = ""
+	var availZone string = ""
+
+	requestAnnotations := req.Machine.Spec.NodeTemplateSpec.ObjectMeta.Annotations
+
+	if _, ok := requestAnnotations[awsPlacementAffinity]; ok && requestAnnotations[awsPlacementAffinity] != "" {
+		affinity = requestAnnotations[awsPlacementAffinity]
+	}
+	if _, ok := requestAnnotations[awsPlacementAvailabilityZone]; ok && requestAnnotations[awsPlacementAvailabilityZone] != "" {
+		availZone = requestAnnotations[awsPlacementAvailabilityZone]
+	}
+	if _, ok := requestAnnotations[awsPlacementTenancy]; ok && requestAnnotations[awsPlacementTenancy] != "" {
+		tenancy = requestAnnotations[awsPlacementTenancy]
+	}
+
+	var placementobj *ec2.Placement = &ec2.Placement{}
+
+	if tenancy != "" {
+		placementobj.Tenancy = &tenancy
+	}
+	if affinity != "" {
+		placementobj.Affinity = &affinity
+	}
+	if availZone != "" {
+		placementobj.AvailabilityZone = &availZone
+	}
+
+	if *placementobj == (ec2.Placement{}) {
+		return nil
+	}
+
+	return placementobj
+}
+
```

Comment:
I see there is a scope to write the `getPlacementObj` function in a cleaner way. Recommending here. Just see if we can adopt this way

```suggestion

//returns Placement Object required in ec2.RunInstancesInput
func getPlacementObj(req *driver.CreateMachineRequest) *ec2.Placement {

	var placementobj = &ec2.Placement{}

	requestAnnotations := req.Machine.Spec.NodeTemplateSpec.ObjectMeta.Annotations

	if affinity, ok := requestAnnotations[awsPlacementAffinity]; ok && affinity != "" {
		placementobj.Affinity = &affinity
	}
	if availZone, ok := requestAnnotations[awsPlacementAvailabilityZone]; ok && availZone != "" {
		placementobj.AvailabilityZone = &availZone
	}
	if tenancy, ok := requestAnnotations[awsPlacementTenancy]; ok && tenancy != "" {
		placementobj.Tenancy = &tenancy
	}

	if *placementobj == (ec2.Placement{}) {
		return nil
	}

	return placementobj
}

```

Reply:
yes sure